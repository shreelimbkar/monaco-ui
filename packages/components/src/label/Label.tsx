import React from 'react';

import BoxProps from '../types/BoxProps';
import { LabelStyles } from './LabelStyles';

const Label: React.FC<BoxProps> = (props) => {
  const { children } = props;
  const ref = React.useRef(null);
  return (
    <LabelStyles as="label" ref={ref} {...props}>
      {children}
    </LabelStyles>
  );
};

export default Label;
