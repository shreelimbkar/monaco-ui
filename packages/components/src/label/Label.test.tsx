import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Label from '.';

afterEach(cleanup);

describe('Label Component', () => {
  test('Renders Correctly', () => {
    const { asFragment } = render(<Label />);
    expect(asFragment()).toMatchSnapshot();
  });
});
