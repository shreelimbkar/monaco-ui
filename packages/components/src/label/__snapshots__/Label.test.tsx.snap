// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label Component Renders Correctly 1`] = `
<DocumentFragment>
  <label
    class="sc-bdvvtL gmAoZd"
  />
</DocumentFragment>
`;
